xarray/core/computation.py

```diff
@@ -579,20 +589,40 @@ def apply_variable_ufunc(
                 "``.load()`` or ``.compute()``"
             )
         elif dask == "parallelized":
-            input_dims = [broadcast_dims + dims for dims in signature.input_core_dims]
             numpy_func = func
 
+            if dask_gufunc_kwargs is None:
+                dask_gufunc_kwargs = {}
+
             def func(*arrays):
-                return _apply_blockwise(
-                    numpy_func,
-                    arrays,
-                    input_dims,
-                    output_dims,
-                    signature,
-                    output_dtypes,
-                    output_sizes,
-                    meta,
-                )
+                import dask.array as da
+
+                try:
+                    res = da.apply_gufunc(
+                        numpy_func, str(signature), *arrays, **dask_gufunc_kwargs,
+                    )
+                # catch error that was not raised in earlier versions of `xr.apply_ufunc`
+                except ValueError as e:
+                    if "with different chunksize present" in str(e):
+                        warnings.warn(
+                            str(e) + " This will throw an error in the future."
+                        )
+                        res = da.apply_gufunc(
+                            numpy_func,
+                            str(signature),
+                            *arrays,
+                            **dask_gufunc_kwargs,
+                            allow_rechunk=True,
+                        )
+                    else:
+                        raise
```

Comment:
```suggestion
                res = da.apply_gufunc(
                    numpy_func, str(signature), *arrays, **dask_gufunc_kwargs,
                )
```

Reply:
I think @shoyer suggested to just error here. (Also my mistake for leading you down this path.)